Typings: use Renderers for renderers of ReactMarkdownProps

b6pzeusbc54tvhw5jgpyw8pwz2x6gs opened the pull request:
I think `renderers` of `ReactMarkdownProps` should use interface `Renderers`

rexxars commented:
Are there any specific reason why that would be better? I am not super familiar with typescript yet.

There has been a lot of changes in this lately, maybe @jessepinho or @ihsw has any input?

jessepinho commented:
@b6pzeusbc54tvhw5jgpyw8pwz2x6gs Is [`Renderer<T>`](https://github.com/rexxars/react-markdown/blob/c25a51660a35f3a7b3d58e2a6b6eeffde91fe0a9/index.d.ts#L91) roughly the same as `ReactType`? If so, this should be fine. Also, have you tested your change?

ihsw commented:
@rexxars it could be better as it more clearly conveys that the return type can be a `string` or `Renderer`.

`Renderer<T>` is roughly the same as `ReactType<T>` though.

b6pzeusbc54tvhw5jgpyw8pwz2x6gs commented:
@rexxars @jessepinho @ihsw 
Thank you for your discussion. I am also not familiar with typescript yet.

The [builtin default renderers.js][renderes] has **string** return type. I think the default renderers and user customized renderers both should use the same type.

[renderes]: https://github.com/rexxars/react-markdown/blob/7cddebfeb1891fdcceff3571f65dabe53241e146/src/renderers.js

jessepinho commented:
Again, have you tested your change on a custom renderer? (And you mentioned you're not familiar with TypeScript — are you actually using this change? If not, let's not change something that isn't broken. 😄 )

b6pzeusbc54tvhw5jgpyw8pwz2x6gs commented:
@jessepinho Yes, I tested my change.

It is my custom renderers,
```typescript
const customRenderers: ReactMarkdown.Renderers = {
  linkReference: (props) => {
    const videoUrl = props.href
    const thumbnailUrl = props.children[0].props.src
    if (
      (lookupMime(videoUrl) || '').split('/')[0] === 'video'
      && (lookupMime(thumbnailUrl) || '').split('/')[0] === 'image'
    ) {
      // we use custom video player on this case
      return <span style={{ display: 'none' }}/>
    }
    return <a {...props}/>
  },
}
```

Before the change, I encountered the following error, but after the change, the error disappears.
<img width="1024" alt="screen shot 2019-02-06 at 1 15 13 am" src="https://user-images.githubusercontent.com/15520015/52287172-09159080-29ad-11e9-99cf-54b8aab7e411.png">

Rocamonde commented:
Dear all, @rexxars 

What's going on with this? This still seems to be an issue today. I cannot seem to use this project with TypeScript. Is this fix planned for any upcoming release?

Best

wooorm commented:
Hi, thanks for your patience!
I’m gonna go with https://github.com/remarkjs/react-markdown/pull/428, which will result in lots of merge conflicts that it’s probably better to close this and take a new look if there still is something missing.
